src/components/common/ui/form/numbered-text-area-mui/NumberedTextArea.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { useRef } from 'react';
+import {
+  Box,
+  FormControl,
+  FormHelperText,
+  OutlinedInput,
+  useMediaQuery,
+} from '@mui/material';
+import { SxProps, Theme } from '@mui/material/styles';
+import { useField } from 'formik';
+
+import LineNumbers from '@/components/common/ui/form/numbered-text-area-mui/components';
+import { transformValue } from '@/components/common/ui/form/numbered-text-area-mui/utils';
+import mergeSx from '@/lib/utils/MergeSxStylesUtil';
+import theme from '@/styles/theme';
+
+import * as styles from './NumberedTextArea.styles';
+
+interface NumberedTextAreaProps {
+  name: string;
+  placeholder?: string;
+  disabled?: boolean;
+  showRemark?: boolean;
+  sx?: SxProps<Theme>;
+}
+
+const MAX_LENGTH = 2000;
+
+const NumberedTextArea: React.FC<NumberedTextAreaProps> = ({
+  name,
+  placeholder,
+  disabled = false,
+  showRemark = false,
+  sx,
+}) => {
+  const [field, { touched, error }] = useField(name);
+  const lineRef = useRef<HTMLDivElement>(null);
+  const isMobile = useMediaQuery(theme.breakpoints.down('mobileMedium'));
+  const state = touched && error ? 'error' : 'default';
+
+  field.value = transformValue(field.value);
+
+  const handleScroll = e => {
+    lineRef.current.scrollTop = e.target.scrollTop;
+  };
+
+  return (
+    <Box sx={mergeSx(styles.wrapper, sx)}>
+      <FormControl sx={styles.formControl(state, disabled)} disabled={disabled}>
+        <LineNumbers value={field.value} ref={lineRef} />
+
+        <OutlinedInput
+          {...field}
+          sx={styles.input}
+          rows={isMobile ? 7 : 10}
```

Comment:
Fix magic numbers please. Maybe extract them like you did with `MAX_LENGTH`

Reply:
![image](https://github.com/fictadvisor/fictadvisor-web/assets/89925253/fb91f67b-6bdb-4877-abfa-9ffcc1a4005a)